tests/casp/stack_admin.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Start admin node (velum)
+# Maintainer: Martin Kravec <mkravec@suse.com>
+
+use base "opensusebasetest";
+use strict;
+use testapi;
+use lockapi;
+
+sub run() {
+    # Admin node needs long time to start web interface
+    # Wait in loop until velum is available until controller node can connect
+    my $timeout   = 240;
+    my $starttime = time;
+    while (script_run 'curl -I localhost | grep velum') {
+        my $timerun = time - $starttime;
+        if ($timerun < $timeout) {
+            record_soft_failure "Velum starting for " . $timerun . "s";
```

Comment:
We should avoid having a record_soft_failure with no bug reference. If you encountered this already then reference an existing bug report. If you did not yet encounter it then just die here with a TODO-entry to replace it by a record_soft_failure